HW2/P2/P2.py

```diff
@@ -20,19 +21,23 @@
     dest = np.random.randint(1000, size=1000000).astype(np.int32)
 
     total = orig_counts.sum()
+    ##print total, len(orig_counts)
 
     # serial move
     counts = orig_counts.copy()
+    ##print counts.max(), len(counts)
     with Timer() as t:
         move_data_serial(counts, src, dest, 100)
     assert counts.sum() == total, "Wrong total after move_data_serial"
     print("Serial uncorrelated: {} seconds".format(t.interval))
+    ##print counts.sum(), len(counts)
     serial_counts = counts.copy()
 
     # fine grained
     counts[:] = orig_counts
+
     with Timer() as t:
-        move_data_fine_grained(counts, src, dest, 100)
+        move_data_fine_grained(counts, src, dest, 100, threads)
```

Comment:
I don't think it is a good idea to change the function definition. It is fine in an individual project like this, but if you are working with collaborators, your teammates may not be happy about this.